typescript-test-samples/step-functions-local/src/tests/sfnLocal/step-functions-local.test.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+/**
+ * step-functions-local.test.ts
+ *
+ * Unit tests for the lead generation Step Function leveraging the Step Functions Local docker image.
+ *
+ */
+
+import { beforeAll, afterAll, describe, it, expect, jest } from '@jest/globals';
+import { GenericContainer, Wait, StartedTestContainer }  from 'testcontainers';
+import { 
+    SFNClient, 
+    StartExecutionCommand, 
+    StartExecutionCommandOutput, 
+    CreateStateMachineCommand, 
+    CreateStateMachineCommandOutput, 
+    GetExecutionHistoryCommand, 
+    HistoryEvent 
+} from "@aws-sdk/client-sfn";
+import { StepFunctionsConstants } from './step-functions-constants';
+
+// Modify if default timeout isn't enough
+// jest.setTimeout(10000);
+
+/**
+ * Unit Tests
+ *
+ * This is the set of unit tests focused on the lead generation system, testing
+ * the step function state machine. 
+ * 
+ */
+
+describe('lead-generation', () => {
+    describe('step-function-local', () => {
+
+        let container: StartedTestContainer;
+        let sfnClient: SFNClient;
+        let createStateMachineResponse: CreateStateMachineCommandOutput;
+        let stateMachineArn: string;
+
+        beforeAll(async () => {
+
+            // Launch StepFunctionLocal with testcontainers
+            container = await new GenericContainer('amazon/aws-stepfunctions-local')
+                .withExposedPorts(8083)
+                .withCopyFilesToContainer([{ 
+                    source: StepFunctionsConstants.mockFileHostPath, 
+                    target: StepFunctionsConstants.mockFileContainerPath
+                  }])
+                .withEnvironment({ SFN_MOCK_CONFIG: StepFunctionsConstants.mockFileContainerPath })
+                .withDefaultLogDriver()
+                .withWaitStrategy(Wait.forLogMessage("Starting server on port 8083"))
+                .start();
+
+            // Set up Step Function client with test container URL
+            sfnClient = new SFNClient({ 
+                endpoint: `http://${container.getHost()}:${container.getMappedPort(8083)}`
+            });                
+
+            // Important to avoid non-deterministic behavior
+            await sleep(2);         
```

Comment:
Could we add a line about why this avoids non-determinism? Are we waiting for the container to spin up?

Reply:
Updated in new pull request to be submitted. Yes, we are waiting for the container to spin up.